view/adminhtml/templates/boltpay/js/replacejs.phtml

```diff
@@ -376,21 +376,41 @@ require([
         },
 
         success: function (transaction, callback) {
+
+            /**
+             * Success transaction handler.
+             * Sets the success url for the non-preauth flow.
```

Comment:
Isn't this being called if `is_pre_auth` is true?

Reply:
@matt-thomason 

If pre-auth is true then the redirect_url, which is created and returned in result of create order call, is implicitly opened from the checkout, when passed callback() is executed. And it redirects to the success page if passes validation.

In no pre-auth we need to call order/save, which returns the success url that we save as a property of the callbacks object. on the success order/save call, as the last step, after storing the success_url, the callback() is called. One of the things it does is closing the checkout modal. In the close event handler we check for the success_url and open the success page.

To sumarize:
pre-auth - checkout implicitly does the success page redirect
no pre-auth - the redirect is called explicitly in the plugin code

The later was already done in pre-auth and it was redundant.